unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/AvatarLODController/AvatarLODController.cs

```diff
@@ -154,6 +155,21 @@ private void SetAvatarFeatures(bool newSSAOEnabled, bool newFacialFeaturesEnable
             }
         }
 
+        public void UpdateImpostorTint(float sqrDistanceToMainPlayer)
+        {
+            // We need to correct distance values as the distance to main player is squared 
+            float minImpostorTintDistance = AvatarRendererHelpers.IMPOSTOR_TINT_MIN_DISTANCE * AvatarRendererHelpers.IMPOSTOR_TINT_MIN_DISTANCE;
+            float maxImpostorTintDistance = AvatarRendererHelpers.IMPOSTOR_TINT_MAX_DISTANCE * AvatarRendererHelpers.IMPOSTOR_TINT_MAX_DISTANCE;
+
+            float initialStep = Mathf.Max(minImpostorTintDistance, sqrDistanceToMainPlayer);
+            float tintStep = Mathf.InverseLerp(minImpostorTintDistance, maxImpostorTintDistance, initialStep);
+            float tintValue = Mathf.Lerp(AvatarRendererHelpers.IMPOSTOR_TINT_NEAREST_BLACKNESS, AvatarRendererHelpers.IMPOSTOR_TINT_FAREST_BLACKNESS, tintStep);
+            Color newColor = Color.Lerp(Color.white, Color.black, tintValue);
+            newColor.a = Mathf.Lerp(AvatarRendererHelpers.IMPOSTOR_ALPHA_NEAREST_VALUE, AvatarRendererHelpers.IMPOSTOR_ALPHA_FAREST_VALUE, tintStep);
```

Comment:
Uhmm, yeap, I just ran other calculations and you are right, this solution is no good =(.

I reverted the changes and I'm back at using the `Mathf.Sqrt(sqrtDistance)`, because getting the regular distance in `ComposeLODControllersSortedByDistance()` would defeat the purpose of using `SqrMagnitude` (and I would have to modify the tuple returned to return an object or something, as now would need 3 properties).

In the current state we only run the `Mathf.Sqrt()` for impostors and not for other avatars.

If you have another idea I'm open tu suggestions of course.

Reply:
We are using SqrDistance because it's cheaper to calculate but if we have to calculate the square root anyway for the tint, then it's less performant because we square some values.

So let's just use distance everywhere.

Reply:
I'm inclined to just use Distance for simplicity sake's because we will have (in the worst case, the one that matters) more impostors than avatars.

But I kinda agree with your approach, is a CPU heavy algorithm and we should squeeze as much performance as possible.

Good job!

Reply:
When we have all the solutions we are implementing joined together in master, after new profiling if we find issues with these calculations we can have another try at changin the approach =)